Remotion/Data/DomainObjects/Persistence/Rdbms/SqlServer/Parameters/SqlTableValuedParameterValue.cs

```diff
@@ -82,5 +82,84 @@ IEnumerator<SqlDataRecord> IEnumerable<SqlDataRecord>.GetEnumerator ()
   }
 
   public int Count => _records.Count;
+
+  public bool Equals (SqlTableValuedParameterValue? other)
```

Comment:
Is there benefit to having this on the production code? if you need to compare in the tests, just add a helper there.

On production, we are makign a promise that it's a well implemented and helpful api. if there is no use case for comparing the Values, we should not do it, given the amount of prod-code this generates.

Reply:
The test infrastructure checks whether the parameter value equals the expected parameter value - see `SqlProviderGeneratedSqlTestHelper.CheckCommand()`.
If we want to get rid of the `Equals()` implementation in the production code, we would have to inject an IEqualityComparer<SqlTableValuedParameterValue> with each table-valued parameter, and use that for equality assertions instead of `Is.EqualTo()`.
Is that what you had in mind?

Reply:
I was thinking, we could just add a check-method in the test fixture that compares the expected and actual ones and gives Assert-errors inline. Similiar to CheckClassDefinition etc.

Reply:
ah, I understand. this is needed because we check if the parameter value matches the mock expectations.

I would suggest, to make the test easier/diagnostics better, that we match any tvp parameter, back up the value in the callback and assert it at the end of the test. that way, we get good diangostics if there's an error in the parameter's contents and not simply a failed mock.

Reply:
I still don't understand. `SqlProviderGeneratedSqlTestHelper.CheckCommand()` is used by 19 tests, only 4 of which use a TVP - the others also need to check for equality of the parameter value. If we want to continue using a common helper method, we have to make an equality comparison for parameter values of type `SqlTableValuedParameterValue`.